### PR TITLE
Adorsk/behaviors as constructor opts

### PR DESCRIPTION
**Pull request by adorsk**
PR for #2386 .

Additional specs added to test/unit/behaviors.spec.js .

### CODE REVIEW

File: src/behaviors.js [hunk 1]
```diff
@@ -20,7 +20,8 @@ Marionette.Behaviors = (function(Marionette, _) {
 
     // Behaviors defined on a view can be a flat object literal
     // or it can be a function that returns an object.
-    behaviors = Behaviors.parseBehaviors(view, behaviors || _.result(view, 'behaviors'));
+    behaviors = Behaviors.parseBehaviors(view, behaviors ||
+        _.result(view.options, 'behaviors') || _.result(view, 'behaviors'));
```
Comment (jasonLaster) [inline src/behaviors.js]: Hmm maybe Marionette.getOption

Comment (adorsk) [inline src/behaviors.js]: Roger that, updated in PR to use Marionette.getOption .

**Comment by jasonLaster** [comment]
Two things:
1. This PR should be reopened against next
2. The behavior tests are bloody. Try running the tests with grunt test. Or open SpecRunner in the browser

**Comment by paulfalgout** [comment]
The latest issue is trying to use `_.result`  with `getOption`

functionally you probably want something like: `Mn._getValue(Mn.getOption(view, 'behaviors'), view);`

But I think this might be cleaner if we utilize the 2nd argument of Behaviors.  So essentially what is now:
```js
function Behaviors(view, behaviors) {

    if (!_.isObject(view.behaviors)) {
      return {};
    }

    // Behaviors defined on a view can be a flat object literal
    // or it can be a function that returns an object.
    behaviors = Behaviors.parseBehaviors(view, behaviors || _.result(view, 'behaviors'));
```
 would become:

```js
function Behaviors(view, behaviors) {

    if (!_.isObject(behaviors)) {
      return {};
    }

    behaviors = Behaviors.parseBehaviors(view, behaviors);
```
And then over in the View https://github.com/marionettejs/backbone.marionette/blob/next/src/abstract-view.js#L22 you'd have :
```js
var behaviors = Marionette._getValue(this.getOption('behaviors'), this);
this._behaviors = Marionette.Behaviors(this, behaviors);
```

**Comment by adorsk** [comment]
@jasonLaster: Roger that, I'll close here and re-open against next.

@paulfalgout: Thx for the tips, your suggestion looks good to me. I'll update and use it when I re-open the PR against next.

**Pull request closed without merging**